### DISCUSSION TITLE
Question on location of last_played_list

### DISCUSSION THREAD

**Discussion by john-** [jcorporation/myMPD#839] [Q&A] [answered]
myMPD version: 10.0.0

**Describe the question**

The release notes for version 10.0.0 indicate that the format of last_played file has changed.  The `sed` one liner indicates that this file is stored in `/var/lib/mympd/state/default`.  On my install (v9.4.1) I do not have the `default` subdirectory.   Should I add this directory?   Or should target directory be where the other state file are?

**Comment by jcorporation** [comment, marked as answer]
The default directory will be created at startup of the new version. You can create it by hand in the directory where your other state files are.

**Reply by john-** [reply]
As you indicate the default directory was created after the first startup.  I did the conversion after that.

Thanks for the release!

**Reply by john-** [reply]
Everything seems to me be working fine in my setup with 10.0.0 but I am curious about the last_played_list file.  This file doesn't seem to get updated after a song has played.  What is the purpose of this file?

Also, can I delete /var/lib/mympd/state/last_played after doing the conversion?

**Reply by jcorporation** [reply]
`/var/lib/mympd/state/last_played` can be deleted after conversion. The new last_played_list file is not updated on each song change. Recent last played songs are kept in memory and are written in batch jobs to disc, after 10 songs and on exit.

**Reply by john-** [reply]
Thanks!
